chain/x/zoracle/handler_test.go

```diff
@@ -299,10 +297,18 @@ func TestRequestWithPrepareGasExceed(t *testing.T) {
 	dataSource := keep.GetTestDataSource()
 	keeper.SetDataSource(ctx, 1, dataSource)
 
-	// set prepare gas to 3 (not enough for using) then it occurs error.
-	msg := types.NewMsgRequestData(1, calldata, 2, 2, 100, 3, 1000000, sender)
+	msg := types.NewMsgRequestData(1, calldata, 2, 2, 100, 1000000, sender)
 
+	gasConsumedBeforeHandle := ctx.GasMeter().GasConsumed()
 	got := handleMsgRequestData(ctx, keeper, msg)
+	gasConsumedAfterHandle := ctx.GasMeter().GasConsumed()
+
+	// the code's prepare function is expected to consume 103 gas
+	ctx = ctx.WithGasMeter(sdk.NewGasMeter(gasConsumedAfterHandle - gasConsumedBeforeHandle - 103))
+
+	// handleMsgRequestData will panic when the gas limit doesn't enough to consume
+	require.Panics(t, func() { handleMsgRequestData(ctx, keeper, msg) }, "The code did not panic")
```

Comment:
In my opinion this code panic because the gas doesn't enough to consume for execution not run prepare owasm code. 

*Note, I think if gas prepare is not enough it will return error not panic code.